arduinolux/src/Debug/DebugTest.cpp

```diff
@@ -0,0 +1,14 @@
+#include"../StringConverter.h"
+#include <string>
+#include <iostream>
+
+int main() {
+	std::string testString = "Hello there world!";
+	do {
+		char* convertedString = StringConverter::ExtractCharArrayFromString(testString);
+
+		std::cout << convertedString << "\n";
```

Comment:
*facepalm* I'm surprised I missed that... of course the data would be junk when I return stack data! Guess it hasn't quite sunk in that arrays aren't in heap-space by default in C++...

Still, I did notice one potential upside (or also potential downside) to the more verbose converter I wrote. c_str returns const char*, which would make it hard to validate it for compatibility with the translux. Keeping the data mutable with ExtractCharArrayFromString as is may actually be worth the extra clock cycles in this case (though I very well may be wrong, or we could do the validation to the std string before converting it to char*). Thoughts? 

Either way, the function was fun just to write.